File: solved_class3/2606.py
def virus(index):
    global visit
    for i in v[index]:
        if i not in visit:
            visit.add(i)
            virus(i)
    return


nodeNum = int(input())
edgeNum = int(input())
visit = set()
v = [[] for _ in range(nodeNum + 1)]

for _ in range(edgeNum):
    edge1, edge2 = map(int, input().split())
    v[edge1].append(edge2)
    v[edge2].append(edge1)

virus(1)
if nodeNum == 1:
    print(0)
else:
    print(len(visit) - 1)

File: solved_class3/2606.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

vector<vector<int>> node;
set<int> visit;

void dfs(int index)
{
    for (int i = 0; i < node[index].size(); i++)
        if (!visit.count(node[index][i]))
        {
            visit.insert(node[index][i]);
            dfs(node[index][i]);
        }
    return;
}

int main()
{
    int nodeNum, edgeNum;
    cin >> nodeNum >> edgeNum;
    node.resize(nodeNum + 1);

    while (edgeNum--)
    {
        int edge1, edge2;
        cin >> edge1 >> edge2;
        node[edge1].push_back(edge2);
        node[edge2].push_back(edge1);
    }
    dfs(1);
    visit.erase(1);
    cout << visit.size();
    return 0;
}